Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -280,16 +280,15 @@ FieldValue FieldValue::ArrayValue(std::vector<FieldValue>&& value) {
   return result;
 }
 
-FieldValue FieldValue::ObjectValue(
-    const std::map<std::string, FieldValue>& value) {
-  std::map<std::string, FieldValue> copy(value);
-  return ObjectValue(std::move(copy));
+FieldValue FieldValue::ObjectValueFromMap(const ObjectValue::Map& value) {
```

Comment:
Do we need this rename? Are we going to have other overloads that conflict?

Reply:
It already conflicts with the ObjectValue struct. (I would've thought the compiler could disambiguate between the two, but apparently not.)

(More context: In the other languages, this method lives in ObjectValue itself and is just called FromMap.)

Reply:
We could do the same here.